### ISSUE TITLE
LabelEncoder not using corretly  Unique labels with Tensors

### ISSUE THREAD

**Issue by tarbaig** [albumentations-team/albumentations#2192]
## Describe the bug

```albumentations.core.utils.LabelEncoder```'s fit method tries to get unique labels using ```set(...)```, ```set(Tensor)``` however returns a ```Set``` instance that contains all emlements of the tensor.

This is the case sind PyTorch tensors are objects, and when passed directly to set(), their object identity (memory address) is used for comparison, not their numeric values. This means two tensors with identical values but different memory addresses will be considered distinct.

I can create an MR if interest exists, happy to help.

### To Reproduce
```python
from albumentations.core.utils import LabelEncoder
import torch
labels = torch.tensor([1,1,1,1,2])
lenc = LabelEncoder()
lenc.fit(labels)
print(lenc.classes_)
# prints:
#  {tensor(1): 0, tensor(1): 1, tensor(1): 2, tensor(1): 3, tensor(2): 4}
# which is most likely not what is intended
```
 
### Expected behavior

Use numeric values of tensor labels for uniqueness test.

### Actual behavior

Uses object identity for uniqueness test, leading to objects t labeled wit the same value being treated as different.

**Comment by tarbaig** [comment]
albumentations = "^1.4.22"

**Comment by ternaus** [comment]
What is the use case when you pass Torch Tensor as labels?

Right now Albumentations has pytorch only as optional dependency, as some people use the library in Keras / Tensorflow.

I can make LabelEncoder work with labels in tensor format, but it will force to make optional depencency pytorch a mandatory one.

On the other hand, it is possible to convert labels to numpy arrays and convert them back to torch tensors after the transform. It is two extra lines of code, which is incoenvenient, but I do not see any other way how to solve the issue.

If you know a way to handle this corner case in the library without introducing pytorch as a depencency - I am all attention.

**Comment by tarbaig** [comment]
Interestingly code that fails due to this in ```1.4.22``` works in ```1.1.0```.

Checking if an object is a tensor without importing torch could be done like this:
```python
def is_pytorch_tensor(obj):
    return obj.__class__.__module__ == 'torch' and obj.__class__.__name__ == 'Tensor'
```

Since pyton is duck typed, working with the 2-3 needed tensor funtions to achieve the same can than ( in the relevant cases ) be done without importing torch.

My usecase was, that I inherited a codebase uning albumentations which failed due to this. I downgraded the library, now it works.

**Comment by ternaus** [comment]
Thanks, will try this.

**Comment by ternaus** [comment]
It is interesting corner case, but as everything works for you, will close this issue for now.
